scripts/linux/pkg/linux-packaging.md

```diff
@@ -0,0 +1,88 @@
+# RavenDB installation from package on Linux
+
+## What the package does on install
+
+- scatter the files in the default locations
+
+- install systemd service
+
+- add soft links to binaries
+
+- add `ravendb` user and group
+
+- set file permissions properly
+
+- setcap on the Raven.Server
+
+
+## Supported use cases
+
+### Ran under systemd
+The default is when RavenDB is used as a systemd daemon and controlled with systemd.
+The following directory layout (initialized by the package installation process) and permissions is used in this scenario:
+
+#### Filesystem locations and permissions
+
+- settings - `/etc/ravendb/settings.json`
+
+- data - `/var/lib/ravendb/data`
+
+- security (e.g. certificate)  - `/var/lib/ravendb/security`
+
+- logs  - `/var/log/ravendb/logs`
+
+- audit logs  - `/var/log/ravendb/audit`
+
+- binaries (current Server dir) - `/usr/local/ravendb`
+
+- rvn link - `/usr/bin/rvn -> /usr/local/ravendb/rvn`
+
+- ravendb - `/usr/bin/ravendb -> /usr/local/ravendb/Raven.Server` 
+
+- systemd unit file - `/etc/systemd/system/ravendb.service -> /usr/local/ravendb/ravendb.service`
+
+#### Permissions
+
+| Location | Owner | Group | Permissions |
+| - | - | - | - |
+| `/usr/local/ravendb/server` | root | ravendb | 550 |
+| `/usr/bin/rvn -> /usr/local/ravendb/rvn` | root | ravendb | 550 |
+| `/usr/bin/ravendb -> /usr/local/ravendb/Raven.Server` | root | ravendb | 550 |
+| `/etc/ravendb/settings.json` | root | ravendb | 440 |
+| `/var/lib/ravendb/data` | root | ravendb | 770 |
+| `/var/lib/ravendb/security` | root | ravendb | 770 |
+| `/var/log/ravendb/logs` | root | ravendb | 770 |
+| `/var/log/ravendb/audit` | root | ravendb | 770 |
+
+## Running as a regular user
+
+The idea is for the users to be able to run multiple instances of RavenDB on a multi-user system, keeping the data and the settings in their home directory.
+When run as a regular user we start on a random port without settings, producing RavenData dir (and all the other dirs) in the working dir.
+
+Default settings should include proper retention policies for all logs.
+Turn on audit logs.
+
+## Package lifecycle
+
+### Installation
+
+When installed and run the first time we start in `SetupMode` `Initial`.
+
```

Comment:
Need to consider this scenario. Assume that I'm using SSH to connect to the server.
How do I do the setup?

We can either make it work via command line browser (doubtful) or need to emit the right command to show SSH tunneling and connect to remotely to the local port.

Reply:
CLI browser is out of question. They are not that powerful to run Studio or even the setup part - too complex. As far as tunneling is concerned it's probably their thing to do it - we can put something in the docs and in the package readme, but I would not invest there. 

Main question here is how do we set it up when having just the terminal. Ideas for the future:

- a CLI setup wizard script, sending Setup Wizard payloads via curl, doing LE and puff - it's there. It's feasible for sure. 

- simpler, no wizard, but elaborate `ravendb-setup` for which you'd pass few to multiple switches (I mean options elaboration level certbot) and they would init the settings file
```
ravendb-setup \
-o raven-setup-package.zip \
--license ./license.json \
--accept-eula \
--agree-le-tos \
--node-ip 1.2.3.4 \
--node-ip 2.2.3.4 \
--node-ip 1.6.6.7 \
--domain ravendb.com
# probably few more here
```
or
```
ravendb-setup --localhost --unsecured --yes-i-accept-its-unsecured
```
People are getting used to this kind of commands nowadays - mentioned `certbot` or `docker` have this kind of interface and it works for them.

Without a tool like this we can:
- start in setup as we do by default - document accessing it the first time if set up on a "headless" machine
- start unsecured on localhost, which is not ideal

Thoughts?

Reply:
the `ravendb-setup` command might be the best option, yes. But also needs to be interactive, I think

That said, I think that setting up the system to allow SSH tunneling is easiest to do and give us all the benefits.
We just need to provide the command and ensure that the user is aware of that.

By default, we use dynamic port, which isn't ideal, though.

Reply:
We don't have a good place to show such command when installing the package - server is not running yet and we don't know the port. Even if we did, we don't know how does the user connect to this server in the first place (the key/password, tunneling the ssh itself through some bastion) since we would need to mimic that. I don't think we're able to do that.

solutions:
- `ravendb-setup` mentioned above
- include `open-ravendb-setup-to-the-world.sh` in the package
- add explicit info in the docs on how to setup on headless/ssh-only machines - we already have some piece here sort of https://ravendb.net/docs/article-page/5.1/Csharp/start/installation/setup-examples/aws-linux-vm

Reply:
Yes, we will pretty much have to run a fixed port. We can use a high one, which is unlikely to be busy.

Reply:
Added a message like this during the install:
![image](https://user-images.githubusercontent.com/1541805/112005197-d3977700-8b22-11eb-8936-4db57d5e8a18.png)

Reply:
Please add the `ssh` command to do the port forwarding as well here.

Reply:
Since we don't know the external hostname the best I think we can do is something like this:
![image](https://user-images.githubusercontent.com/1541805/112822173-e14f7e00-9087-11eb-9fea-1f78ccbd277a.png)

Does it work for you?

Reply:
Can't we use: `dig +short myip.opendns.com @resolver1.opendns.com ` ?

Reply:
OK, sure. Might cover most cases. Here's what I have now:
```
$ sudo dpkg -i ravendb_4.2.112-0_amd64.deb
Selecting previously unselected package ravendb.
(Reading database ... 70102 files and directories currently installed.)
Preparing to unpack ravendb_4.2.112-0_amd64.deb ...
Unpacking ravendb (4.2.112-0) ...
Setting up ravendb (4.2.112-0) ...
Adjust createdump binary permissions...
Adjust Raven.Debug binary permissions...
###
#
#  Please navigate to http://127.0.0.1:53700 in your web browser to complete setting up RavenDB.
#  If you set up the server through SSH, you can tunnel RavenDB setup port and proceed with the setup on your local.
#
#  For public IP:   ssh -N -L localhost:8080:localhost:53700 ubuntu@54.198.78.116
#
#  For internal IP: ssh -N -L localhost:8080:localhost:53700 ubuntu@ip-10-0-2-76
#
###
Processing triggers for libc-bin (2.27-3ubuntu1) ...
```

Sufficient? Still don't know if somebody logged in with password or key so we might be wrong with regards to auth method.

Reply:
Looks really nice, yes.

Reply:
It is enough that we provide this, the actual auth is on them.

We can't handle everything.

For example, we might not be able to handle it if they force to bind to localhost and use another port, etc. 
Not on us in this case.